src/middleware/console-status-verification.ts

```diff
@@ -69,70 +68,90 @@ async function consoleStatusVerificationMiddleware(request: express.Request, res
 		return;
 	}
 
-	// * This is kinda temp for now. Needs to be redone to handle linking this data to existing 3DS devices in the DB
-	// TODO - 3DS consoles are created in the NASC middleware. They need special handling to link them up with the data in the NNID API!
-	if (request.certificate.consoleType === 'wiiu') {
-		const certificateDeviceID: number = parseInt(request.certificate.certificateName.slice(2), 16);
+	let device = await Device.findOne({
+		serial: serialNumber,
+	});
 
-		if (deviceID !== certificateDeviceID) {
-			// TODO - Change this to a different error
-			response.status(400).send(xmlbuilder.create({
-				error: {
-					cause: 'Bad Request',
-					code: '1600',
-					message: 'Unable to process request'
-				}
-			}).end());
+	const certificateHash = crypto.createHash('sha256').update(request.certificate._certificate).digest('base64');
```

Comment:
Future thing to think about maybe: sha512? These are 32-bit systems (3ds and Wii U) so idk if it will impact that but 64-bit servers could probably handle it

Reply:
That would break compatibility with older documents. sha256 is sufficient in this case